src/boot/efi/stub.c

```diff
@@ -91,6 +91,36 @@ static void combine_measured_flag(int *value, int measured) {
         *value = *value < 0 ? measured : *value && measured;
 }
 
+static void measure_section(
+                long unsigned int section_base,
+                size_t section_size,
+                UnifiedSection section,
+                int *sections_measured) {
+        assert(section_base);
+        assert(section_size);
+        assert(sections_measured);
+
+        /* First measure the name of the section */
+        bool m = false;
+        (void) tpm_log_ipl_event_ascii(
+                        TPM2_PCR_KERNEL_BOOT,
+                        POINTER_TO_PHYSICAL_ADDRESS(unified_sections[section]),
+                        strsize8(unified_sections[section]), /* including NUL byte */
+                        unified_sections[section],
+                        &m);
+        combine_measured_flag(sections_measured, m);
+
+        /* Then measure the data of the section */
+        m = false;
+        (void) tpm_log_ipl_event_ascii(
+                        TPM2_PCR_KERNEL_BOOT,
+                        section_base,
+                        section_size,
+                        unified_sections[section],
+                        &m);
+        combine_measured_flag(sections_measured, m);
+}
```

Comment:
so please don't measure the ucode stuff with this. Please use tagged event measurement instead, see how this is done for dt. i.e. make up a new tag like DEVICETREE_ADDON_EVENT_TAG_ID does it, and then measure it with that.

the measurement of sections we do so far we must retain for compat, but it's actually very much not fun to dissect later from the measurement logs, hence we shouldn't add more to that.